### PR TITLE
New Pattern: API Gateway Save to DynamoDB and Stream to Lambda with CDK which combine both TypeScript for CDK and Python Lambda Code

### PR DESCRIPTION
**Pull request by nguyendinhthi0705**
*Issue #, if available:*

*Description of changes:*
 Implement a new pattern with API Gateway to insert into DynamoDB stream to trigger a Lambda

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by julianwood** [comment]
I'm getting a `Missing Authentication Token` when testing this with the initial :
image

**Comment by nguyendinhthi0705** [comment]
> I'm getting a `Missing Authentication Token` when testing this with the initial DDB put: image

Hi @julianwood ,
I just update the readme file correct instruction.
The API should be append awsomedynamodb , somthing like https://${API_ID}.execute-api.${REGION_NAME}.amazonaws.com/prod/awsomedynamodb
And you also does not need id on url for a post

File: apigw-rest-api-dynamodb-lambda-cdk/README.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+![Architecture Diagram](img/concept.png)
+## Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+    ``` 
+    git clone https://github.com/aws-samples/serverless-patterns
+    ```
+1. Change directory to the pattern directory:
+    ```
+    cd apigw-rest-api-dynamodb-lambda-cdk/cdk
```
Comment (julianwood) [inline apigw-rest-api-dynamodb-lambda-cdk/README.md:27]: Should be `cd apigw-rest-api-dynamodb-lambda-cdk`

File: apigw-rest-api-dynamodb-lambda-cdk/README.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    ```
+
+1. Install dependencies
+    ```
+    npm install
+	cd src
```
Comment (julianwood) [inline apigw-rest-api-dynamodb-lambda-cdk/README.md:33]: Should be `cd ..\src`

File: apigw-rest-api-dynamodb-lambda-cdk/README.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+	pip install aws-xray-sdk --target .
+	pip install aws_lambda_powertools --target .
+    ```
+
+1. Deploy the stack to your default AWS account and region. The output of this command should give you the HTTP API URL.
+    ```
```
Comment (julianwood) [inline apigw-rest-api-dynamodb-lambda-cdk/README.md:39]: Add `cd ..`

File: apigw-rest-api-dynamodb-lambda-cdk/README.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+	```
+	* Select **POST** as the HTTP method from the drop-down list to the left of the address bar.
+	* Choose the **Body** tab. Choose **raw** and select **JSON** from the drop-down list. Enter the following into the text box: 
+	```
+	{
+		"id":"14",
+    	"name":"Do some thing",
+    	"description":"Do some thing",
+    	"customer":"John"
+	}
```
Comment (julianwood) [inline apigw-rest-api-dynamodb-lambda-cdk/README.md:70]: Th items you are adding don't seem relevant to what you query in the testing

File: apigw-rest-api-dynamodb-lambda-cdk/README.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+	* Choose **Send** to submit the request and receive a "200 OK" response with a list of the matching results. Example: 
+	```
+		{
+			"Count": 1,
+			"Items": [
+				{
+					"pk": {
+						"S": "foo"
+					},
+					"data": {
```
Comment (julianwood) [inline apigw-rest-api-dynamodb-lambda-cdk/README.md:93]: The `pk` and `data` are not items you are putting in the previous section.  Please check your intputs

File: apigw-rest-api-dynamodb-lambda-cdk/README.md (hunk 2 above)
Comment (nguyendinhthi0705) [reply apigw-rest-api-dynamodb-lambda-cdk/README.md:33]: Hey, I am not sure about this. It should be: cd src

File: apigw-rest-api-dynamodb-lambda-cdk/README.md (hunk 3 above)
Comment (nguyendinhthi0705) [reply apigw-rest-api-dynamodb-lambda-cdk/README.md:39]: Done

File: apigw-rest-api-dynamodb-lambda-cdk/README.md (hunk 5 above)
Comment (nguyendinhthi0705) [reply apigw-rest-api-dynamodb-lambda-cdk/README.md:93]: Updated

File: apigw-rest-api-dynamodb-lambda-cdk/README.md (hunk 4 above)
Comment (nguyendinhthi0705) [reply apigw-rest-api-dynamodb-lambda-cdk/README.md:70]: I am not sure about this, you can put anything for testing, right ?

**Comment by nguyendinhthi0705** [comment]
Hi @julianwood J
I fixed from your comment except 2 items. Please review. So far I think it ís good, just update content Readme ?

File: apigw-rest-api-dynamodb-lambda-cdk/README.md (hunk 2 above)
Comment (nguyendinhthi0705) [reply apigw-rest-api-dynamodb-lambda-cdk/README.md:33]: yes, just cd src . I tested on Mac

File: apigw-rest-api-dynamodb-lambda-cdk/README.md (hunk 4 above)
Comment (nguyendinhthi0705) [reply apigw-rest-api-dynamodb-lambda-cdk/README.md:71]: Yes, you can put anything. But although DynamoDB support schemeless, we actually should control scheme in real product

**Comment by nguyendinhthi0705** [comment]
Hi @julianwood ,
Please review and let me know if you have any concern!
